src/controller/model/endpoint.ts

```diff
@@ -70,10 +96,56 @@ interface ConfiguredReporting {
     reportableChange: number,
 }
 
-interface PendingRequest {
-    /* eslint-disable-next-line @typescript-eslint/no-explicit-any*/
-    func: () => Promise<any>, resolve: (value: any) => any, reject: (error: any) => any,
-    sendWhen: 'active' | 'fastpoll', expires: number, lastError: Error
+/* eslint-disable-next-line @typescript-eslint/no-explicit-any*/
+export class Request<Type = any> {
```

Comment:
I propose to move this to a separate file, `endpoint.ts` is getting quite big. What about `src/controller /helpers/request.ts`?

Reply:
sounds good! I moved it there, together with the defaultSendPolicy as a static member